File: front/src/app/atividade/modelo/dtos/atividadeDTO.ts
export class AtividadeDTO {
  titulo: string;
  descricao: string;
  valorCredito: number;
  valorDebito: number;
  frequencia: string;
  idResponsavel: string;

  constructor(titulo: string, descricao: string, valorCredito: number, valorDebito: number, frequencia: string, idResponsavel: string) {
    this.titulo = titulo;
    this.descricao = descricao;
    this.valorCredito = valorCredito;
    this.valorDebito = valorDebito;
    this.frequencia = frequencia;
    this.idResponsavel = idResponsavel;
  }
}

File: backend/src/main/java/com/mang/funtask/dominio/dto/request/AtividadeDTO.java
package com.mang.funtask.dominio.dto.request;

import com.mang.funtask.dominio.enums.Frequencia;
import com.mang.funtask.dominio.modelos.Atividade;
import java.util.UUID;

public record AtividadeDTO(String titulo, String descricao, Double valorCredito, Double valorDebito,
                           Frequencia frequencia, UUID idResponsavel) {

  public AtividadeDTO(Atividade atividade) {
    this(atividade.getTitulo(), atividade.getDescricao(), atividade.getValorCredito(), atividade.getValorDebito(), atividade.getFrequencia(), atividade.getIdResponsavel());
  }
}
